src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,181 @@
+package util;
+
+import backend.Logic;
+import backend.github.ApiQuotaInfo;
+import org.apache.logging.log4j.Logger;
+import ui.UI;
+import util.events.RefreshTimerTriggeredEvent;
+import util.events.RefreshTimerTriggeredEventHandler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages the API quota usage and periodic refreshes of the data store.
+ */
+public class ApiQuotaManager {
+
+    private static final Logger logger = HTLog.get(ApiQuotaManager.class);
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    private static final int API_QUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period.
+     */
+    private static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    /**
+     * The uninitialized value for uninitialized variables.
+     */
+    private static final int UNINITIALIZED = -1;
+
+    /**
+     * The amount of the available remaining API requests.
+     */
+    private int remainingApiRequests = UNINITIALIZED;
+
+    /**
+     * The number of API calls used in most recent refresh.
+     */
+    private int apiCallsUsedInPreviousRefresh = 0;
```

Comment:
They don't get initialised at the same time, so i didn't implement ApiUsageInfo.